responsibleai/responsibleai/feature_metadata.py

```diff
@@ -82,4 +98,28 @@ def __eq__(self, other_feature_metadata) -> bool:
             self.categorical_features == \
             other_feature_metadata.categorical_features and \
             self.dropped_features == \
-            other_feature_metadata.dropped_features
+            other_feature_metadata.dropped_features and \
+            self.time_series_id_features == \
+            other_feature_metadata.time_series_id_features
+
+    @staticmethod
+    def _validate_columns(
+            column_purpose: str,
+            column_names: List[str],
+            feature_names: List[str]):
+        """Ensure the provided column is present in the dataset.
+
+        :param column_purpose: The purpose the column fulfills in the dataset.
+        :type column_purpose: str
+        :param column_names: List of column names to validate.
+        :type column_names: List[str]
+        :param feature_names: List of features in the input dataset.
+        :type feature_names: List[str]
+        """
+        if column_names is None:
+            return
+        for column_name in column_names:
+            if column_name not in feature_names:
```

Comment:
nit: maybe make feature_names a set before this loop?  otherwise it's O(n^2), since it's a list, no?  with feature_names as a set it will be O(n) to do this check.  Probably this won't be too long regardless since the number of features here is usually going to be pretty small though.